### PR TITLE
Facet / Aggregation support

### PR DESCRIPTION
**Pull request by josbeir**
Hi,

I was wondering if there is currently any way to add facets / aggregations to the query. Cant seem to find this in the Query class.

### CODE REVIEW

**Comment by josbeir** [comment]
not really expecting this to be merged but this approach works fine for me for the time being :-)

File: src/Query.php [hunk 1]
```diff
@@ -298,6 +300,44 @@ public function query($matcher)
     }
 
     /**
+     * Add a facet to the elastic query object
+     *
+     * @param  array|Elastica\Facet\AbstractFacet $facet One or multiple facets
+     * @return $this;
+     */
+    public function facet($facet)
```
Comment (lorenzo) [inline src/Query.php]: I would not include this at all. Facets is a deprecated concept in latests versions of elastic search

**Comment by lorenzo** [comment]
@josbeir Other than my comment, the code looks fine. Having tests would be amazing

File: src/Query.php [hunk 2]
```diff
@@ -298,6 +300,44 @@ public function query($matcher)
...
+        }
+
+        return $this;
+    }
+
+    /**
+     * Add an aggregation to the elastic query object
+     *
+     * @param  array|Elastica\Facet\AbstractAggregation $aggregation One or multiple facets
+     * @return $this;
```
Comment (ADmad) [inline src/Query.php]: There should be no `;` at the end :)

File: src/Query.php (hunk 1 above)
Comment (josbeir) [inline src/Query.php]: indeed, i'm not even using it myself :-) i'll remove it

**Comment by josbeir** [comment]
@lorenzo removed facet() and the semicolon, tests will come a bit later because of the standard company pressure. stand by :-)

File: src/Query.php [hunk 3]
```diff
@@ -298,6 +299,25 @@ public function query($matcher)
     }
 
     /**
+     * Add an aggregation to the elastic query object
+     *
+     * @param  array|Elastica\Facet\AbstractAggregation $aggregation One or multiple facets
+     * @return $this
+     */
+    public function aggregation($aggregation)
```
Comment (lorenzo) [inline src/Query.php]: Would you mind changing this to `aggregate()`? I can add the test cases for you :)

**Comment by josbeir** [comment]
@lorenzo renamed Query::aggregation to Query::aggregate

**Comment by lorenzo** [comment]
Thanks!

**Pull request merged**